[Term Entry] Python Strings .isalnum() (#3295)
Codecademy/docs: content/python/concepts/strings/terms/isalnum/isalnum.md

---
Title: '.isalnum()'

> Description: 'Takes a string and returns True if all the  characters are alphanumeric (A-Z, a-z, 0-9). Otherwise it returns False if the string contains any non-alphanumeric characters (@#$%-*)

Reviewer: ```suggestion
Description: 'Returns True if all the characters in a given string are alphanumeric.'
```

Author: Updated ✅

Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Methods'
  - 'Functions'
CatalogContent:
  - 'learn-python-3'
  - 'paths/analyze-data-with-python'
---

> The `.isalnum()` string method takes in a string and returns `True` if all the  characters are alphanumeric (A-Z, a-z, 0-9). Otherwise it returns `False` if the string contains any [non-alphanumeric](https://buildvirtual.net/a-guide-to-non-alphanumeric-characters/) characters (@#$%-*).

Reviewer: ```suggestion
The **`.isalnum()`** string method takes in a string and returns `True` if all the characters are alphanumeric (`A-Z`, `a-z`, `0-9`). Otherwise it returns `False` if the string contains any non-alphanumeric characters (e.g., `@ # $ % - *`).
```

## Syntax

> ```py

Reviewer: ```suggestion
```pseudo
```

Author: Update ✅

string.isalnum()
```

> 

Reviewer: There should be a line here such as-
"The method does not have any parameters. "

Author: Line added ✅

## Example

> Use `.isalnum()` to check if all the  characters are alphanumeric:

Reviewer: ```suggestion
The examples below use `.isalnum()` to check if all the characters are alphanumeric:
```

```py
my_string_1 = "string2023"

print(my_string_1.isalnum())
# Output: True

my_string_2 = "@lpha"

print(my_string_2.isalnum())
# Output: False

> ```

Reviewer: For concepts that involve lots of examples we often comment the output. But, for most term entries we usually use a ```shell block.

## Codebyte Example

> Use `.isalnum()` to check if a `string` contains only alphanumeric letters or not:

Reviewer: ```suggestion
The code below is runnable and uses the `isalnum()` method:
```

Author: Updated ✅.

> ```codebyte/python

Author: Removed the second one ✅.
